pkg/vault/contracts/AddUnbalancedLiquidityViaSwapRouter.sol

```diff
@@ -170,39 +169,39 @@ contract AddUnbalancedLiquidityViaSwapRouter is RouterQueries, IAddUnbalancedLiq
                     tokenIn: tokens[adjustableTokenIndex],
                     tokenOut: hookParams.operationParams.exactToken,
                     amountGivenRaw: swapAmount,
-                    limitRaw: hookParams.operationParams.maxAdjustableAmount - amountsIn[adjustableTokenIndex],
+                    limitRaw: limit,
                     userData: hookParams.operationParams.swapUserData
                 })
             );
 
             amountsIn[adjustableTokenIndex] += swapAmountIn;
-            if (amountsIn[exactTokenIndex] >= swapAmountOut) {
-                amountsIn[exactTokenIndex] -= swapAmountOut;
-            } else {
-                amountsIn[exactTokenIndex] = 0;
-                amountsOut[exactTokenIndex] = swapAmountOut - amountsIn[exactTokenIndex];
-            }
+            amountsIn[exactTokenIndex] -= swapAmountOut;
         } else if (amountsIn[exactTokenIndex] < hookParams.operationParams.exactAmount) {
             uint256 swapAmount = hookParams.operationParams.exactAmount - amountsIn[exactTokenIndex];
+
             (, uint256 swapAmountIn, uint256 swapAmountOut) = _vault.swap(
                 VaultSwapParams({
                     kind: SwapKind.EXACT_IN,
                     pool: hookParams.pool,
                     tokenIn: hookParams.operationParams.exactToken,
                     tokenOut: tokens[adjustableTokenIndex],
                     amountGivenRaw: swapAmount,
-                    limitRaw: 0,
+                    limitRaw: limit,
                     userData: hookParams.operationParams.swapUserData
                 })
             );
 
             amountsIn[exactTokenIndex] += swapAmountIn;
-            if (amountsIn[adjustableTokenIndex] >= swapAmountOut) {
-                amountsIn[adjustableTokenIndex] -= swapAmountOut;
-            } else {
-                amountsOut[adjustableTokenIndex] = swapAmountOut - amountsIn[adjustableTokenIndex];
-                amountsIn[adjustableTokenIndex] = 0;
-            }
+            amountsIn[adjustableTokenIndex] -= swapAmountOut;
+        }
+
+        if (amountsIn[exactTokenIndex] != hookParams.operationParams.exactAmount) {
+            revert AmountInDoesNotMatchExact(amountsIn[exactTokenIndex], hookParams.operationParams.exactAmount);
+        } else if (amountsIn[adjustableTokenIndex] > hookParams.operationParams.maxAdjustableAmount) {
+            revert AmountInAboveMaxAdjustableAmount(
+                amountsIn[adjustableTokenIndex],
+                hookParams.operationParams.maxAdjustableAmount
+            );
```

Comment:
There's no way to trigger these in practice because the swap will fail with the swap limit before reaching here right?

How about not using the limit in the swap? These error messages are actually better in the context of an `addLiquidity` operation, so maybe it's better to let the swap through and fail here (even though it'll consume some more gas in the failing case). And with that we could actually test these conditions.

WDYT?